File: Source/Templates/Projects/VBfull/MyRibbon.vb
Imports System.Runtime.InteropServices
Imports ExcelDna.Integration.CustomUI
Imports VBfull.My.Resources

<ComVisible(True)>
Public Class MyRibbon
    Inherits ExcelRibbon

    Public Overrides Function GetCustomUI(RibbonID As String) As String
        Return RibbonResources.Ribbon
    End Function

    Public Overrides Function LoadImage(imageId As String) As Object
        ' This will return the image resource with the name specified in the image='xxxx' tag
        Return RibbonResources.ResourceManager.GetObject(imageId)
    End Function

    Public Sub OnButtonPressed(control As IRibbonControl)
        System.Windows.Forms.MessageBox.Show("Hello!")
    End Sub

End Class

File: Source/Templates/Projects/CSfull/MyRibbon.cs
using System.Runtime.InteropServices;
using ExcelDna.Integration.CustomUI;

namespace CSfull;

[ComVisible(true)]
public class MyRibbon : ExcelRibbon
{
    public override string GetCustomUI(string RibbonID)
    {
        return RibbonResources.Ribbon;
    }

    public override object? LoadImage(string imageId)
    {
        // This will return the image resource with the name specified in the image='xxxx' tag
        return RibbonResources.ResourceManager.GetObject(imageId);
    }

    public void OnButtonPressed(IRibbonControl control)
    {
        System.Windows.Forms.MessageBox.Show("Hello!");
    }
}
